src/kernel_1.jl

```diff
@@ -20,3 +20,52 @@ function kernel_1(
 
   return g
 end
+
+function get_next_type(T::DataType)
+    (T== UInt8) && return  (UInt16,8)
+    (T== UInt16) && return (UInt32,16)
+    (T== UInt32) && return (UInt64,32)
+    (T== UInt64) && return (UInt128,64)
+    error("Overflow")
+end
+
+function kernel_1_new(
+  ij::Matrix{T}
+  ) where T<:Integer
+
+  # Creating Graph
+  nv = maximum(ij) # Number of vertices
+  g = Graph{T}(nv) # Creating a graph with `nv` number of vertices
+  T_upper, sftamt = get_next_type(T)
+  v = Vector{T_upper}()
```

Comment:
What if this were a `SharedVector` with a `@parallel` for loop down below? (How can we parallelize kernel 1?)

Reply:
`v` works exactly as you mentioned
```
julia> using Graph500

julia> ij = kronecker_generator(16,16)
2×1048576 Array{UInt32,2}:
 0x00009d9e  0x00007fd0  0x00001456  …  0x0000db67  0x00003718  0x00008aec
 0x0000958c  0x00005ccb  0x00004e82     0x00001110  0x000031a5  0x00007c76

julia> kernel_1_new(ij)
  0.022737 seconds
{65536, 909678} undirected simple UInt32 graph
```
Also the execution time for the loop below is just .02sec for graph with `SCALE=16`
Since you were planning for running final benchmarks on `SCALE=14` it will make `kernel1` slower